common/test/acceptance/pages/studio/settings_certificates.py

```diff
@@ -171,37 +172,75 @@ def click_confirmation_prompt_primary_button(self):
         self.wait_for_ajax()
 
 
-class Certificate(object):
+class CertificateSectionPage(CertificatesPage):
     """
-    Certificate page object wrapper
+    CertificateSectionPage is the certificate section within Certificates page, There might be multiple certificates
+    in a Certificates Page so this section object can be used to used to identify unique certificate and apply
+    operations on it.
     """
 
-    def __init__(self, page, prefix, index):
-        self.page = page
+    def __init__(self, container_certificates_page, prefix, index):
+        """
+        Initialize CertificateSection Page
+
+        :param container_certificates_page: Container Page Object of the certificate section
+        :param prefix: css selector of the container element
+        :param index: index of section in the certificate list on the page
+
+        :return:
+        """
         self.selector = prefix + ' .certificates-list-item-{}'.format(index)
         self.index = index
+        self.container = container_certificates_page
+
+        super(CertificateSectionPage, self).__init__(
+            container_certificates_page.browser,
+            **container_certificates_page.course_info
+        )
+
+    def is_browser_on_page(self):
+        """
+        Verify that the browser is on the page and it is not still loading.
+        """
+        EmptyPromise(
+            lambda: self.container.q(css=".certificates").present,
+            'Wait for Certificate Section to load.'
+        ).fulfill()
+        return True
+
+    @property
+    def url(self):
+        """
+        Construct a URL to the page section within the certificate page.
+        """
+        # The url is same as CertificatePage url for now.
+        return self.container.url
 
     ################
     # Helpers
     ################
 
+    @unguarded
     def get_selector(self, css=''):
         """
         Return selector fo certificate container
         """
         return ' '.join([self.selector, css])
 
-    def find_css(self, css_selector):
+    @unguarded
+    def q(self, **kwargs):
         """
-        Find elements as defined by css locator.
+        Override q method to search elements contained within the Certificate Section only.
```

Comment:
Why is this necessary? I would thikn you could just use the same `q` method that you would get for free in the PageObject.